magnolia/shared/src/test/scala-dotty/zio/config/magnolia/AutomaticConfigSpec.scala

```diff
@@ -27,12 +27,16 @@ object AutomaticConfigTest extends ZIOSpecDefault {
 }
 
 object AutomaticConfigTestUtils {
+
+  @discriminator("type")
   sealed trait Credentials
 
   case class Password(value: String) extends Credentials
 
   case class Token(value: String) extends Credentials
 
+  case object InstanceProfile extends Credentials
```

Comment:
This PR enables this use case - discriminated sealed traits having case objects as children.